pkg/apis/pipeline/v1beta1/resource_types_validation.go

```diff
@@ -25,29 +25,19 @@ import (
 	"knative.dev/pkg/apis"
 )
 
-func (tr *TaskResources) Validate(ctx context.Context) *apis.FieldError {
-	if tr == nil {
-		return nil
-	}
-	if err := validateTaskResources(tr.Inputs, "inputs"); err != nil {
-		return err
-	}
-	if err := validateTaskResources(tr.Outputs, "outputs"); err != nil {
-		return err
+func (tr *TaskResources) Validate(ctx context.Context) (errs *apis.FieldError) {
```

Comment:
pointer named return value 😞, could this result in reference to `nil` on line 30 or 31?

Reply:
yes and no, that's the beauty of it :stuck_out_tongue:  (and this part of `knative.dev/pkg`)
See https://pkg.go.dev/knative.dev/pkg@v0.0.0-20200915170341-1d4e654a38f0/apis#FieldError.Also (implementation is [here](https://github.com/knative/pkg/blob/1d4e654a38f0/apis/field_error.go#L111)).

```
func (fe *FieldError) isEmpty() bool {
	if fe == nil {
		return true
	}
	return fe.Message == "" && fe.Details == "" && len(fe.errors) == 0 && len(fe.Paths) == 0
}
```

It won't ever panic, because it knows how to act on a `nil` struct  :wink: